### PR TITLE
Convert Rechunker to GraphStage

### PR DESCRIPTION
**Pull request by johanandren**
Supersedes #20481
Refs #20288

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala [hunk 1]
```diff
@@ -4,56 +4,71 @@
...
 import scala.util.Random
 
 class FramingSpec extends StreamSpec {
 
   val settings = ActorMaterializerSettings(system)
   implicit val materializer = ActorMaterializer(settings)
 
-  class Rechunker extends PushPullStage[ByteString, ByteString] {
-    private var rechunkBuffer = ByteString.empty
+  class Rechunker extends GraphStage[FlowShape[ByteString, ByteString]] {
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: oh so it's only in tests

**Comment by ktoso** [comment]
LGTM

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala (hunk 1 above)
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala]: yup. nothing to fuss about, but nice to get it off off the list of prs

**Comment by ktoso** [comment]
Thanks @pchordia15 and @johanandren !

**Pull request merged**